lib/utils/ignore.ts

```diff
@@ -200,6 +200,41 @@ export interface FileStats {
 	stats: Stats;
 }
 
+/**
+ * Create a list of files for the filesystem subtree rooted at
+ * projectDir, excluding entries for directories themselves.
+ * @param projectDir Source directory (root of subtree to be listed)
+ */
+async function listFiles(projectDir: string): Promise<string[]> {
+	const dirs: string[] = [];
+	const files: string[] = [];
+	dirs.push(projectDir);
+	async function walk(currentDirs: string[]): Promise<string[]> {
+		if (!currentDirs.length) {
+			return files;
+		}
+
+		const foundDirs: string[] = [];
+
+		await Promise.all(
+			currentDirs.map(async (dir) => {
+				const _files = await fs.readdir(dir, { withFileTypes: true });
+				for (const entry of _files) {
+					const fpath = `${dir}/${entry.name}`;
+					if (entry.isDirectory()) {
+						foundDirs.push(fpath);
+						dirs.push(fpath);
+					} else {
+						files.push(fpath);
+					}
+				}
+			}),
+		);
+		return walk(foundDirs);
+	}
+	return walk([projectDir]);
```

Comment:
```suggestion
		return await walk(foundDirs);
	}
	return await walk([projectDir]);
```
It is not incorrect to leave out `await` here, but I understand that adding `await` improves debug tracing (easier to follow stack traces) when the returned promised is rejected, as the rejection gets associated with the line of code where `await` is used. But this is just me parroting what I was told (or read) before, as I've never tested it myself to confirm.

In the case of the inner `walk` call specifically, as it is recursive, I wonder if adding `await` would reduce parallelisation by also reducing the depth of the recursive stack (also reducing memory usage). In which case, the `await` keyword would not be just a "good practice" matter but a performance matter, controlling a tradeoff between memory and speed.